test/framework/src/main/java/org/elasticsearch/indices/SystemIndexThreadPoolTestCase.java

```diff
@@ -88,9 +89,14 @@ private void assertThreadPoolsBlocked() {
         assertThat(e2.getMessage(), startsWith("rejected execution of ActionRunnable"));
         var e3 = expectThrows(
             SearchPhaseExecutionException.class,
-            () -> client().prepareSearch(USER_INDEX).setQuery(QueryBuilders.matchAllQuery()).get()
+            () -> client().prepareSearch(USER_INDEX)
+                .setQuery(QueryBuilders.matchAllQuery())
+                // Don't let the test framework set maxConcurrentShardRequests randomly, because the
```

Comment:
Could we set it to a random value other than 1?

Reply:
Is `usually() ? SearchRequest.DEFAULT_MAX_CONCURRENT_SHARD_REQUESTS : randomIntBetween(2, 10)` too elaborate?

Reply:
That looks ideal to me